packages/dds/tree/src/test/simple-tree/objectNode.spec.ts

```diff
@@ -21,6 +21,7 @@ describe("ObjectNode", () => {
 				toString: schemaFactory.optional(schemaFactory.number),
 			}) {}
 			{
+				// @ts-expect-error TODO
```

Comment:
thats odd. Does this have to do with the field having a special name failing to shadow correctly when its optional maybe?

Reply:
Doesn't seem to be the case. Updating the field to be `required` yields the same error.
```shell
Type 'number' is not assignable to type '(() => string) & number'
```

Reply:
I take that back - the error is different, but it errors either way. Error for `required`:

```shell
Type 'number' is not assignable to type '((() => string) & number) | (number & (() => string))'.
```

Reply:
After simplifying the type expression for `InsertableObjectFromSchemaRecord`, TypeScript no longer errors when the field is `required`, but it continues to error when the field is `optional` with:
```shell
error TS2322: Type 'number' is not assignable to type '((() => string) & number) | (number & (() => string))'.
```

Reply:
The TypeScript update seems to have resolved this :)